### ISSUE TITLE
Tooling for libraries and git support

### ISSUE THREAD

**Issue by karlmutch** [karlmutch/bump-ver#1]
As a software contributor
I want to enabled library access to the CI/CD style tooling
In order that scripting using golang based CD/CI can be supported

For information about using Go as a scripting language please read, https://blog.cloudflare.com/using-go-as-a-scripting-language-in-linux/

This work is a precursor to being able to easily and quickly implement simple build tooling without complex yaml based configuration or DSLs when simple go based scripts will work.

Basically tired of bash, yaml, complex secrets management for hosted solutions such as Travis, and stack overflow searches when simple engineering would suffice. (rant rant)
